sign_language_datasets/datasets/bsl_corpus/bsl_corpus.py

```diff
@@ -0,0 +1,182 @@
+"""bsl_corpus dataset."""
+
+import os
+
+import tensorflow_datasets as tfds
+from tensorflow_datasets.core import download
+from tensorflow_datasets.core import utils
+
+from . import bsl_corpus_utils
+from ...datasets.config import SignDatasetConfig
+
+_DESCRIPTION = """
+A corpus of British Sign Language.
+"""
+
+_CITATION = """\
+@inproceedings{cormier:12033:sign-lang:lrec,
+  author    = {Cormier, Kearsy and Fenlon, Jordan and Johnston, Trevor and Rentelis, Ramas and Schembri, Adam and Rowley, Katherine and Adam, Robert and Woll, Bencie},
+  title     = {From Corpus to Lexical Database to Online Dictionary: Issues in annotation of the {BSL} Corpus and the Development of {BSL} {SignBank}},
+  pages     = {7--12},
+  editor    = {Crasborn, Onno and Efthimiou, Eleni and Fotinea, Stavroula-Evita and Hanke, Thomas and Kristoffersen, Jette and Mesch, Johanna},
+  booktitle = {Proceedings of the {LREC2012} 5th Workshop on the Representation and Processing of Sign Languages: Interactions between Corpus and Lexicon},
+  maintitle = {8th International Conference on Language Resources and Evaluation ({LREC} 2012)},
+  publisher = {{European Language Resources Association (ELRA)}},
+  address   = {Istanbul, Turkey},
+  day       = {27},
+  month     = may,
+  year      = {2012},
+  language  = {english},
+  url       = {https://www.sign-lang.uni-hamburg.de/lrec/pub/12033.pdf}
+}
+"""
+
+_HOMEPAGE = "https://bslcorpusproject.org/"
+
+_VIDEO_RESOLUTION = (640, 360)
+
+_FRAMERATE = 25
+
+
+class BslCorpus(tfds.core.GeneratorBasedBuilder):
+    """
+  DatasetBuilder for BSL corpus dataset.
+
+  The dataset currently loads annotations only, and ignores video files.
+  """
+
+    VERSION = tfds.core.Version('1.0.0')
+    RELEASE_NOTES = {
+        # Third release of annotations:
+        # https://bslcorpusproject.org/wp-content/uploads/Notes-to-the-3rd-release-of-BSL-Corpus-annotations.pdf
+        '3.0.0': 'Third release.',
+    }
+
+    BUILDER_CONFIGS = [
+        SignDatasetConfig(name="default", include_video=False, include_pose=None),
+        SignDatasetConfig(name="annotations", include_video=False, include_pose=None),
+    ]
+
+    def __init__(self, bslcp_username: str, bslcp_password: str, **kwargs):
+
+        super(BslCorpus, self).__init__(**kwargs)
+
+        self.bslcp_username = bslcp_username
+        self.bslcp_password = bslcp_password
+
+    def _make_download_manager(self, download_dir, download_config):
+        """Creates a new download manager object."""
+        download_dir = (
+                download_dir or os.path.join(self._data_dir_root, "downloads"))
+        extract_dir = (
+                download_config.extract_dir or os.path.join(download_dir, "extracted"))
+        manual_dir = (
+                download_config.manual_dir or os.path.join(download_dir, "manual"))
+
+        if download_config.register_checksums:
+            # Note: Error will be raised here if user try to record checksums
```

Comment:
i think its fine, checksums are not mandatory

Reply:
this note is from the original tfds code